### PR TITLE
Signup box code

### PR DESCRIPTION
**Pull request by Jimmy618**
pull request from signupboxCode to develop

### CODE REVIEW

**Comment by Henry-YSLin** [comment]
Need to change "username" back to "name", cause "username" no longer exists

File: src/client/API.js [hunk 1]
```diff
@@ -73,17 +73,17 @@ class API {
...
    */
-  signup(user, email, password) {
-    console.log('signup API parameter', user, email, password);
+  signup(name, email, password) {
+    console.log('signup API parameter', name, email, password);
     return this.request({
       method: 'POST',
       url: '/auth/signup',
-      data: { user, email, password }
+      data: { name, email, password }
```
Comment (Henry-YSLin) [inline src/client/API.js:86]: The issue is here. Putting `name` here is actually a short form of `name: name`, so if you put `user here instead, the client is sending the field as `user`, not `name`

**Review by Henry-YSLin — APPROVED** [review]
The code is working on deployment

**Review by Henry-YSLin — APPROVED** [review]

File: src/client/API.js (hunk 1 above)
Comment (Jimmy618) [reply src/client/API.js:86]: Thank you

**Pull request merged**